### PR TITLE
Fix O(N^2) UNIX domain socket read behavior

### PR DESCRIPTION
**Pull request by santtu**
See issue https://github.com/web3j/web3j/issues/565.

### CODE REVIEW

File: core/src/main/java/org/web3j/protocol/ipc/UnixDomainSocket.java [hunk 1]
```diff
@@ -59,15 +59,15 @@ public void write(String payload) throws IOException {
     @Override
     public String read() throws IOException {
         CharBuffer response = CharBuffer.allocate(bufferSize);
-        String result = "";
+        StringBuffer result = new StringBuffer();
```
Comment (conor10) [inline core/src/main/java/org/web3j/protocol/ipc/UnixDomainSocket.java]: Please can you change this to use StringBuilder?

**Comment by santtu** [comment]
Used StringBuilder instead now. I toyed around increasing the buffer size but did not see any such significant differences in performance between stringbuilder, stringbuffer, and different read buffer sizes to justify any additional changes.

File: core/src/main/java/org/web3j/protocol/ipc/UnixDomainSocket.java (hunk 1 above)
Comment (santtu) [reply core/src/main/java/org/web3j/protocol/ipc/UnixDomainSocket.java]: Done, didn't that change show in the tracker?

**Comment by conor10** [comment]
Thanks!

**Pull request merged**